python/pyarrow/tests/test_array.py

```diff
@@ -3896,3 +3896,26 @@ def test_list_view_slice(list_view_type):
     j = sliced_array.offsets[1].as_py()
 
     assert sliced_array[0].as_py() == sliced_array.values[i:j].to_pylist() == [4]
+
+
+@pytest.mark.parametrize('numpy_dtype', ['>u2', '>i4', '>f8'])
```

Comment:
I think we do have a nightly test runner on a big endian machine, so in that case this hardcoded test will fail? 
So either we have to skip this test in that case, or rewrite the test to work in both cases (numpy has functionality to byteswap the data and change the dtype to the opposite byteorder, so to create non-native data regardless of the platform you are running the test on)

Reply:
Actually, it seems we no longer have such CI coverage because our s390x build was dropped when Travis CI support was dropped .. 

But still, we have regularly reports from people testing on big endian machines, so at least adding a `skipif` based on `sys.byteorder` would be good.

Reply:
@jorisvandenbossche thank you, it is a good point. I've fixed the test to always use non-native byte order